### PR TITLE
chore: Hiero renaming initial

### PR DESCRIPTION
**Pull request by gsstoykov**
**Description**:
This PR introduces Hiero renaming for the C++ SDK.

**Related issue(s)**:

Fixes https://github.com/hiero-ledger/hiero-sdk-cpp/issues/796

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-Copyright 2022 Hedera Hashgraph LLC
+Copyright 2022 Hiero Hashgraph LLC
```
Comment (nathanklick) [inline LICENSE:189]: This is not correct and must be fixed

**Review by nathanklick — CHANGES_REQUESTED** [review]

File: src/sdk/examples/AutoCreateAccountTransferTransactionExample.cc [hunk 2]
```diff
@@ -1,8 +1,8 @@
 /*-
  *
- * Hedera C++ SDK
+ * Hiero C++ SDK
  *
- * Copyright (C) 2020 - 2024 Hedera Hashgraph, LLC
+ * Copyright (C) 2020 - 2024 Hiero
```
Comment (gsstoykov) [inline src/sdk/examples/AutoCreateAccountTransferTransactionExample.cc:5]: I'm not sure about the renaming in the licenses. Can someone give some further details?

File: LICENSE [hunk 3]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-Copyright 2022 Hedera Hashgraph LLC
```
Comment (gsstoykov) [inline LICENSE:189]: I'm not sure about this as well.

File: LICENSE (hunk 1 above)
Comment (gsstoykov) [reply LICENSE]: I put `Copyright 2024 Hiero`

**Review by andrewb1269hg — APPROVED** [review]

**Review by andrewb1269hg — APPROVED** [review]
Reviewed following files:
`.github/workflows/zxc-build-library.yaml`
`CMakeLists.txt`
`LICENSE`
`README.md`
`src/sdk/main/CMakeLists.txt`
`src/sdk/tests/unit/CMakeLists.txt`

**Review by andrewb1269hg — APPROVED** [review]

**Review by rwalworth — APPROVED** [review]

File: LICENSE (hunk 1 above)
Comment (rwalworth) [reply LICENSE]: @nathanklick looks like the file header was fixed https://github.com/hiero-ledger/tsc/issues/64. Could we get an approval on this?

**Review by nathanklick — APPROVED** [review]

**Pull request merged**